eradiate/radprops/particles.py

```diff
@@ -0,0 +1,423 @@
+    )
+
+    top = documented(
+        pinttr.ib(
+            units=ucc.deferred("length"),
+            default=ureg.Quantity(1.0, ureg.km),
+            converter=pinttr.converters.to_units(ucc.deferred("length")),
+            validator=[
+                is_positive,
+                pinttr.validators.has_compatible_units,
+            ],
+        ),
+        doc="Top altitude of the particle layer.\n"
+        "\n"
+        "Unit-enabled field (default: ucc[length]).",
+        type="float",
+        default="1 km.",
+    )
+
+    vert_dist = documented(
+        attr.ib(
+            default={"type": "uniform"},
+            validator=attr.validators.instance_of((dict, ParticleDistribution)),
+        ),
+        doc="Particle number fraction vertical distribution.",
+        type="dict or :class:`ParticleDistribution`",
+        default=":class:`Uniform`",
+    )
+
+    tau_550 = documented(
+        pinttr.ib(
+            units=ucc.deferred("dimensionless"),
+            default=ureg.Quantity(0.2, ureg.dimensionless),
+            validator=[
+                is_positive,
+                pinttr.validators.has_compatible_units,
+            ],
+        ),
+        doc="Extinction optical thickness at the wavelength of 550 nm.\n"
+        "\n"
+        "Unit-enabled field (default: ucc[dimensionless]).",
+        type="float",
+        default="0.2",
+    )
+
+    n_layers = documented(
+        attr.ib(
+            default=None,
+            converter=attr.converters.optional(int),
+            validator=attr.validators.optional(attr.validators.instance_of(int)),
+        ),
+        doc="Number of layers inside the particle layer.\n"
+        "If ``None``, ``n_layers`` is set to a different value based on the "
+        "particle number fraction vertical distribution type (see table below).\n"
+        "\n"
+        ".. list-table::\n"
+        "   :widths: 1 1\n"
+        "   :header-rows: 1\n"
+        "\n"
+        "   * - Distribution type\n"
+        "     - Number of layers\n"
+        "   * - :class:`Uniform`\n"
+        "     - 1\n"
+        "   * - :class:`Exponential`\n"
+        "     - 8\n"
+        "   * - :class:`Gaussian`\n"
+        "     - 16\n"
+        "   * - :class:`Array`\n"
+        "     - 32\n"
+        "\n",
+        type="int",
+        default="``None``",
+    )
+
+    dataset = documented(
+        attr.ib(
+            default=path_resolver.resolve("tests/radprops/rtmom_aeronet_desert.nc"),
+            converter=str,
+            validator=attr.validators.instance_of(str),
+        ),
+        doc="Particles radiative properties data set path.",
+        type="str",
+    )
+
+    def __attrs_post_init__(self):
+        # propagate 'bottom' and 'top' to particle vertical distribution config
+        if isinstance(self.vert_dist, dict):
+            d = self.vert_dist
+            d.update({"bottom": self.bottom, "top": self.top})
+            self.vert_dist = ParticleDistributionFactory.convert(d)
```

Comment:
This conversion operation won't be applied if you pass a `ParticleDistribution`: isn't that a problem?

Reply:
Hmm, yes that's a problem actually